core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/customize-interaction-modal.controller.ts

```diff
@@ -190,10 +188,10 @@ angular.module('oppia').controller('CustomizeInteractionModalController', [
     };
 
     $scope.isSaveInteractionButtonEnabled = function() {
-      return $scope.hasCustomizationArgs &&
-        $scope.StateInteractionIdService.displayed &&
-        $scope.form.schemaForm.$valid &&
-        $scope.getCustomizationArgsWarningsList().length === 0;
+      return !!$scope.hasCustomizationArgs &&
```

Comment:
whats !! ?  Why is that needed?

Reply:
For some reason, this expression was not converting the value to boolean, so the `!!` is to force the parsing to boolean in order to return a boolean.

Reply:
This looks weird. Can you investigate why it wasn't a boolean? maybe it is easier to typecast it rather that doing !! as it is confusing to read.

Reply:
Sure!

Reply:
So, what was the reason for this?

Reply:
After doing some searching, I've found this issue happens because of one of the Javascript's features for logical operators. When using `&&`, it will return the first falsy value. For example:
```js
true && undefined && true; // Instead of returning false, it'll return undefined.
```

That's why I'm converting it to boolean by using `!!`. Some of the values in this expression is not a boolean value, and this turns out to be a tricky behavior. I'm going to convert the entire expression instead of each value to boolean in order to make the code clearer. For more information about this feature, check this [link](https://javascript.info/logical-operators#and-finds-the-first-falsy-value).